src/wallet.js

```diff
@@ -0,0 +1,42 @@
+let secp = require('secp256k1')
+let coins = require('coins')
+let { randomBytes } = require('crypto')
+
+module.exports = function(priv, client) {
+  if (!Buffer.isBuffer(priv) || priv.length !== 32) {
+    throw Error('Private key must be a 32-byte buffer')
+  }
+
+  let creds = {}
+  creds.priv = priv
+  creds.pub = secp.publicKeyCreate(creds.priv)
+  creds.address = coins.secp256k1Account.getAddress({ pubkey: creds.pub })
+
+  return {
+    address: creds.address,
+    priv: creds.priv,
+    pub: creds.pub,
+    getBalance: async function() {
+      try {
+        let balance = await client.state.accounts[creds.address].balance
+        return balance
+      } catch (e) {
```

Comment:
Let's do a check to make sure the error is about the account not existing, otherwise we might hide other errors (e.g. if a node gave us an invalid proof, it would look the same as proving the account has a 0 balance).